src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -455,19 +455,18 @@ private TypeSymbol BindTupleType(TupleTypeSyntax syntax, DiagnosticBag diagnosti
 
             if (hasExplicitNames)
             {
-                // If the tuple type with names is bound in a declaration
-                // context then we must have the TupleElementNamesAttribute to emit
-                if (syntax.IsTypeInContextWhichNeedsTupleNamesAttribute())
-                {
-                    // Report diagnostics if System.String doesn't exist
-                    this.GetSpecialType(SpecialType.System_String, diagnostics, syntax);
+                // If the tuple type with names is bound we must have the TupleElementNamesAttribute to emit
+                // it is typically there though, if we have ValueTuple at all
+                // and we need System.String as well
```

Comment:
This is a preexisting strategy of validating attribute presence in binding and giving errors if attributes are not present.
It might be a good idea to start emitting such diagnostics in emit, but it would be much bigger change.

For now I will follow preexisting pattern and we can consider changing it later for al similar diagnostics.

---
In reply to: [80366117](https://github.com/dotnet/roslyn/pull/14023#discussion_r80366117) [](ancestors = 80366117)